### ISSUE TITLE
How to self-train?

### ISSUE THREAD

**Issue by Outlying3720** [CVMI-Lab/PLA#24]
Thanks a lot for your great work! However, I have some uncertainty regarding the method "Self-Bootstrap with Novel Category Prior" in the paper. 
Does the provided code indeed implement the "with Novel Category Prior"? If it doesn't, could you please share the code that how to implement?

Based on my understanding, the TextSegHead loads CLIP text features for all categories (including the novel classes) through the "set_cls_head_with_text_embed" function. During training, the model appears to be aware of the names of novel categories.

Once again, thank you for your valuable work!

**Comment by Dingry** [comment]
Hi, the text_embed for novel classes is for evaluation only. During training, they are not used. Please refer to this:
https://github.com/CVMI-Lab/PLA/blob/main/pcseg/models/head/text_seg_head.py#L83-L92.

We are sorry that the self-train code is not released since it's just an extension part, not a real "open-word" setting. This part of code needs to be re-organized to be released.

**Comment by Outlying3720** [comment]
Hi, thanks for replying to my issue so promptly!

During training, https://github.com/CVMI-Lab/PLA/blob/main/pcseg/models/head/text_seg_head.py#L83-L92 had make novel scores invisible. But these scores are already calculated with novel class name prior ([Line 71](https://github.com/CVMI-Lab/PLA/blob/main/pcseg/models/head/text_seg_head.py#L71-L71)), so I'm curiously what happens if I drop novel class text embedding (such as all set to 0 or otherfurniture embed) before calculate similarity score?

According to ref 42, I think that you generate pseudo-label of novel classes to finetune the model, is that correct?

**Comment by Dingry** [comment]
Hi, it will have no effect on the result since only scores computed with base categories are kept. The cls_head is just a linear layer. 

Yes. For self-training experiments, we generate pseudo labels to tune the model.

**Comment by Outlying3720** [comment]
I got it and did some tests. It certainly had no different to results.
Thank you for your reply.
